python/ray/train/tensorflow/tensorflow_checkpoint.py

```diff
@@ -48,9 +72,174 @@ def from_model(
+
+            >>> batch_predictor = BatchPredictor.from_checkpoint(
+            ...     result_checkpoint, TensorflowPredictor)
+            >>> batch_predictor.predict(ray.data.range(3))  # doctest: +SKIP
+        """
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, dir_path)
+        checkpoint = cls.from_directory(dir_path)
+        checkpoint._flavor = cls.Flavor.SAVED_MODEL
         return checkpoint
 
-    def get_model_weights(self) -> tf.keras.Model:
-        """Retrieve the model weights stored in this checkpoint."""
-        model_weights, _ = _load_checkpoint(self, "TensorflowTrainer")
-        return model_weights
+    def get_model(
+        self,
+        model_definition: Optional[
+            Union[Callable[[], tf.keras.Model], Type[tf.keras.Model]]
+        ] = None,
+    ) -> tf.keras.Model:
+        """Retrieve the model stored in this checkpoint.
+
+        Args:
+            model_definition: This arg is expected only if the original checkpoint
+                was created via `TensorflowCheckpoint.from_model`.
+
+        Returns:
+            The Tensorflow Keras model stored in the checkpoint.
+        """
+        if self._flavor is self.Flavor.MODEL_WEIGHTS:
+            if not model_definition:
+                raise ValueError(
+                    "Expecting `model_definition` argument when checkpoint is "
+                    "saved through `TensorflowCheckpoint.from_model()`."
+                )
+            model_weights, _ = _load_checkpoint_dict(self, "TensorflowTrainer")
+            model = model_definition()
+            model.set_weights(model_weights)
+            return model
+        else:
+            if model_definition:
+                warnings.warn(
+                    "TensorflowCheckpoint was created from "
+                    "TensorflowCheckpoint.from_saved_model` or "
+                    "`TensorflowCheckpoint.from_h5`, which already contains all the "
+                    "information needed. This means: "
+                    "If you are using BatchPredictor, you should do "
+                    "`BatchPredictor.from_checkpoint(checkpoint, TensorflowPredictor)`"
+                    " by removing kwargs `model_definition=`. "
+                    "If you are using TensorflowPredictor directly, you should do "
+                    "`TensorflowPredictor.from_checkpoint(checkpoint)` by "
+                    "removing kwargs `model_definition=`."
+                )
+            with self.as_directory() as checkpoint_dir:
+                if self._flavor == self.Flavor.H5:
+                    return keras.models.load_model(
+                        os.path.join(checkpoint_dir, self._h5_file_path)
+                    )
+                elif self._flavor == self.Flavor.SAVED_MODEL:
+                    return keras.models.load_model(checkpoint_dir)
+                else:
+                    raise RuntimeError("Unexpected `self._flavor`.")
```

Comment:
This is not supposed to be a user facing error.

Reply:
actually I take it back. It may be user facing. Added description to make it more actionable.